### PR TITLE
Export all markers from the problem window to a file

### LINKED ISSUE #123854: Export all markers from the problem window to a file
I would like to be able to export all markers from the problem window to a file. Here is my scenario: If I do a migration of an existing project to a new version, new technology or something similar, I often end up with a very long list of problems. That is expected and I just need to work through it. But if I decide to share it in a team or just get a better overview, it would be very useful to export the full list to a CSV file, which I can then open in e.g. Excel to sort, filter etc.

### PR DESCRIPTION
**Pull request by tfenster**
This PR fixes #123854

You can test this by opening a workspace with problems and hit the new export button on the top right of the problems view, next to the filter. It should open a save dialog and if you enter a path there, it should export all problems to that file. You can also configure the separator character through the new "Problems: Export Separator" setting (problems.exportSeparator)

### CODE REVIEW

File: src/vs/workbench/contrib/markers/browser/messages.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export default class Messages {
 	public static PROBLEMS_PANEL_CONFIGURATION_TITLE: string = nls.localize('problems.panel.configuration.title', "Problems View");
 	public static PROBLEMS_PANEL_CONFIGURATION_AUTO_REVEAL: string = nls.localize('problems.panel.configuration.autoreveal', "Controls whether Problems view should automatically reveal files when opening them.");
 	public static PROBLEMS_PANEL_CONFIGURATION_SHOW_CURRENT_STATUS: string = nls.localize('problems.panel.configuration.showCurrentInStatus', "When enabled shows the current problem in the status bar.");
+	public static PROBLEMS_PANEL_EXPORT_SEPARATOR: string = nls.localize('problems.panel.export.separator', "Separator character to be used when exporting problems to a file");
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/markers/browser/messages.ts:19]:
```suggestion
	public static PROBLEMS_PANEL_EXPORT_SEPARATOR: string = nls.localize('problems.panel.export.separator', "Separator character to be used when exporting problems to a file.");
```

Comment (tfenster) [reply src/vs/workbench/contrib/markers/browser/messages.ts]: thanks, added the missing dot

**Comment by tfenster** [comment]
@sandy081 May I ask for some feedback on this one? Anything missing or something that I can improve?

**Comment by sandy081** [comment]
I do not think this is a feature that is worth to be added to VS Code out of the box. I would recommend to write an extension using the api - getDiagnostics and implement this feature.

commented the same in original issue - https://github.com/microsoft/vscode/issues/123854

**Pull request closed without merging**